tests-blackbox/common/test-logger.ts

```diff
@@ -0,0 +1,66 @@
+import { ChildProcess } from 'child_process';
+
+type StopFunction = (logs: string, logLine: string) => boolean;
+type FilterFunction = (logLine: string) => boolean;
+
+export class TestLogger {
+	private logs: string;
+	private server: ChildProcess;
+	private stopCondition: StopFunction;
+	private filterCondition?: FilterFunction;
+	private stopped?: boolean;
+	private resolve?: (log: string) => void;
+
+	constructor(server: ChildProcess, stopCondition: string | StopFunction, filter?: string | FilterFunction);
+	constructor(server: ChildProcess, stopCondition: string, filter?: boolean | string | FilterFunction);
+	constructor(server: ChildProcess, stopCondition: string | StopFunction, filter?: boolean | string | FilterFunction) {
```

Comment:
> So far all log entries of the server were captured and checked, which could theoretically lead to false positives, since the conditions in the `expect`s could also be met by other requests

Yes, having `filter` will prevent such theoretical error. However, I think that we should KISS.

Can we simplify `filter` to be just `boolean | string`? 
The usage of boolean to prevent repeated string is a good idea. 👍 
Same for the `stopCondition`, can we revert to `string`?

> I wrapped the tests in a `try...finally` statement to ensure the listeners are also getting cleaned up if the tests fails (e.g. timeout reached). 

Yes, it's no longer required since the extracted implementation in https://github.com/directus/directus/pull/17932/commits/03efdbc5ea0e0728ec59a7cf474bed474cb260f7

Reply:
> Yes, having `filter` will prevent such theoretical error. However, I think that we should KISS.
> 
> Can we simplify `filter` to be just `boolean | string`? The usage of boolean to prevent repeated string is a good idea. 👍 Same for the `stopCondition`, can we revert to `string`?

Agree with KISS, adjusted now 👍 

> Yes, it's no longer required since the extracted implementation in https://github.com/directus/directus/commit/03efdbc5ea0e0728ec59a7cf474bed474cb260f7

What I meant is that it doesn't work anymore, because `cleanup` might not be called if a test fails for whatever reason and the listener remains open. But again I think we can safely disregard this, since it is only the test environment and not the productive application. Just wanted to mention it 😃